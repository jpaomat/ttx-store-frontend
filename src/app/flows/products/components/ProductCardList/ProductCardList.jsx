// import { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { ProductCard } from '../ProductCard/ProductCard';
import './ProductCardList.scss';

export const ProductCardList = ({products}) => {

	return (
		<div className='card-list'>
			{products &&
				products.map((productSelected, i) => (
					<div key={`card-list-${i}`}>
						<Link to={`/details/${productSelected.id}`}>
							<ProductCard {...productSelected} />
						</Link>
					</div>
				))}
		</div>
	);
};

ProductCardList.defaultProps = {
	products : [{}]
};

ProductCardList.propTypes = {
	products: PropTypes.arrayOf(PropTypes.shape({}))
};