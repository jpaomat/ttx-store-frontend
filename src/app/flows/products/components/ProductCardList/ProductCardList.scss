@import '../../../../../assets/scss/';
.card-list {
	@include organizedColumn(space-between);
	a {
		text-decoration: auto;
	}
}

@media screen and (min-width: $break-tablet) {
	.card-list {
		@include organizedMultipleColumns(repeat(2, 1fr));
	}
}

@media screen and (min-width: $break-desktop) {
	.card-list {
		@include organizedMultipleColumns(repeat(4, 1fr));
	}
}
