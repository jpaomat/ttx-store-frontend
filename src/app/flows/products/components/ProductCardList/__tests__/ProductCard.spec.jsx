import Renderer from 'react-test-renderer/shallow';
import { ProductCardList } from '../ProductCardList';
import { BrowserRouter } from 'react-router-dom';

const renderer = new Renderer();

jest.mock('react-router-dom', () => ({
	...jest.requireActual('react-router-dom'),
	useSearchParams: () => [new URLSearchParams({ filter: 'alc' })],
	// useNavigate: () => mockedUsedNavigate
}));

describe('Techtronix component', () => {
	it('should render component', () => {
		const component = renderer.render(
			<BrowserRouter>
				<ProductCardList products={[]} />
			</BrowserRouter>
		);
		expect(component).toMatchSnapshot();
	});
});
