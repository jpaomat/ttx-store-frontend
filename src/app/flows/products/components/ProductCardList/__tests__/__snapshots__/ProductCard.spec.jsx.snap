// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Techtronix component should render component 1`] = `
<div
  className="card-list"
>
  <div>
    <Link
      to="/details/undefined"
    >
      <ProductCard />
    </Link>
  </div>
</div>
`;
